drivers/adc/adc_demo/iio_adc_demo.c

```diff
@@ -43,6 +43,32 @@
 #include "no_os_error.h"
 #include "no_os_util.h"
 #include "iio_adc_demo.h"
+#include "iio.h"
+
+/**
+ * @brief utility function for computing next upcoming channel
+ * @param ch_mask - active channels .
+ * @param last_idx -  previous index.
+ * @param new_idx - upcoming channel index, return param.
+ * @return 1 if there are more channels, 0 if done.
+ */
+static bool get_next_ch_idx(uint32_t ch_mask, uint32_t last_idx,
```

Comment:
Should we do this in `no_os_util.c`?

Reply:
Yes, we can move it seeing how it is already used in 3 different drivers. Thanks!
However, I would prefer doing that in another PR, if that's OK.